File: packs/stack/python-fastapi/scaffold/app/domain/interfaces/note_repository.py
from abc import ABC, abstractmethod

from app.domain.entities.note import Note


class NoteRepository(ABC):
    @abstractmethod
    async def add(self, note: Note) -> Note: ...

    @abstractmethod
    async def get(self, note_id: str) -> Note | None: ...

    @abstractmethod
    async def list_all(self) -> list[Note]: ...

File: packs/data/aws-dynamodb/scaffold/wiring/typescript/src/domain/note-repository.ts
import type { Note } from "./note.js";

/**
 * The storage contract, expressed in entities. A faithful copy of the typescript-node backend's
 * src/domain/note-repository.ts, so the DynamoDB adapter satisfies the exact interface the backend
 * already declares and every implementation behind it is judged by the one contract suite.
 */
export interface NoteRepository {
  add(note: Note): Promise<Note>;
  get(id: string): Promise<Note | null>;
  listAll(): Promise<Note[]>;
}
